OpenRA.Mods.Common/Activities/Resupply.cs

```diff
@@ -76,23 +76,25 @@ public Resupply(Actor self, Actor host, WDist closeEnough, bool stayOnResupplier
 
 		public override bool Tick(Actor self)
 		{
-			// HACK: If the activity is cancelled while we're already resupplying (or about to start resupplying),
-			// move actor outside the resupplier footprint.
-			// TODO: This check is nowhere near robust enough, and should be rewritten.
-			var isCloseEnough = closeEnough < WDist.Zero || (host.CenterPosition - self.CenterPosition).HorizontalLengthSquared <= closeEnough.LengthSquared;
-			if (IsCanceling && isCloseEnough)
-			{
-				foreach (var notifyResupply in notifyResupplies)
-					notifyResupply.ResupplyTick(host.Actor, self, ResupplyType.None);
+			var isHostInvalid = host.Type != TargetType.Actor || !host.Actor.IsInWorld || host.Actor.IsDead;
```

Comment:
Aren't dead actors not in world anyway? That'd mean we can drop the last condition.

Reply:
Actors with `Health` only get disposed/removed when `RemoveOnDeath` is `true`, so the case of an actor being dead but still in world is technically possible (regardless of how few actors might be affected in practice).

Reply:
RemoveOnDeath is a horrible hack that has only stuck around for the LegacyBridge setup. We shouldn’t explicitly account for it anywhere else, and it would be nice if we could migrate RA/TD to the newer bridge code and drop it completely.

Reply:
OK, updated.